audiomentations/augmentations/transforms.py

```diff
@@ -1151,7 +1152,193 @@ def apply(self, samples, sample_rate):
+        :param p: The probability of applying this transform
+        """
+        super().__init__(p)
+
+        self.min_cutoff_freq = min_cutoff_freq
+        self.max_cutoff_freq = max_cutoff_freq
+        if self.min_cutoff_freq > self.max_cutoff_freq:
+            raise ValueError("min_cutoff_freq must not be greater than max_cutoff_freq")
+
+    def randomize_parameters(
+        self, samples: np.array, sample_rate: int = None
+    ):
+        """
+        :params samples: (num_channels, num_samples)
+        """
+        num_samples = samples.shape[-1]
+
+        self.parameters["cutoff_freq"] = np.random.uniform(
+            low=self.min_cutoff_freq,
+            high=self.max_cutoff_freq
+        )
+
+    def apply_transform(self, samples: np.array, sample_rate: int = None):
+        num_channels, num_samples = samples.shape
+
+        if sample_rate is None:
+            sample_rate = self.sample_rate
+
+        cutoffs_as_fraction_of_sample_rate = (
+            self.parameters["cutoff_freq"] / sample_rate
+        )
+
+        samples = high_pass_filter(
+            samples, cutoffs_as_fraction_of_sample_rate.item()
+        )
+
+        return samples
+
+
+class BandPassFilter(BaseWaveformTransform):
+    """
+    Apply band-pass filtering to the input audio.
+    """
+
+    supports_multichannel = True
+    requires_sample_rate = True
+
+    def __init__(
+        self,
+        min_center_freq=100.0,
+        max_center_freq=1000.0,
+        min_q=1.0,
+        max_q=2.0,
+        p=0.5
+    ):
+        """
+        :param min_center_freq: Minimum center frequency in hertz
+        :param max_center_freq: Maximum center frequency in hertz
+        :param min_q: Minimum ratio of center frequency to bandwidth
+        :param max_q: Maximum ratio of center frequency to bandwidth
+        :param p: The probability of applying this transform
+        """
+        super().__init__(p)
+
+        self.min_center_freq = min_center_freq
+        self.max_center_freq = max_center_freq
+        self.min_q = min_q
+        self.max_q = max_q
+        if self.min_center_freq > self.max_center_freq:
+            raise ValueError("min_center_freq must not be greater than max_center_freq")
+        if self.min_q > self.max_q:
+            raise ValueError("min_q must not be greater than max_q")
+
+    def randomize_parameters(
+        self, samples: np.array, sample_rate: int = None
+    ):
+        """
+        :params samples: (num_channels, num_samples)
+        """
+        num_samples = samples.shape[-1]
+
+        self.parameters["center_freq"] = np.random.uniform(
+            low=self.min_center_freq,
+            high=self.max_center_freq
+        )
+        self.parameters["q"] = np.random.uniform(
+            low=self.min_q,
+            high=self.max_q
+        )
+
+    def apply_transform(self, samples: np.array, sample_rate: int = None):
+        num_channels, num_samples = samples.shape
```

Comment:
This will break for 1D numpy arrays and only work for 2D numpy arrays. It looks like you are not using num_channels here - perhaps you can do this instead:

num_samples = samples.shape[-1]

Reply:
I've fixed that.